File: typescript/chatbot.ts
/**
 * Simple chat loop using the Anthropic SDK.
 * Model name is not passed directly to the client constructor, to showcase
 * detection of model usage via code flow.
 */

import Anthropic from "@anthropic-ai/sdk";

const MODEL_KEY = "claude-sonnet-4-20250514";

function init(): { client: Anthropic; model: string } {
  const key = "ANTHROPIC_API_KEY";
  const apiKey = process.env[key];
  if (!apiKey) {
    console.error(`${key} not set in environment.`);
    process.exit(1);
  }
  const client = new Anthropic({ apiKey });
  return { client, model: MODEL_KEY };
}

async function main(): Promise<void> {
  const { client, model } = init();

  console.log("Chat with Claude (type 'exit' to quit).\n");

  const readline = await import("readline");
  const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

  const ask = (): void => {
    rl.question("> ", async (prompt) => {
      if (prompt === "exit") {
        rl.close();
        return;
      }
      try {
        const response = await client.messages.create({
          model,
          max_tokens: 1024,
          messages: [{ role: "user", content: prompt }],
        });
        const text = response.content.find((block) => block.type === "text");
        console.log(text && "text" in text ? text.text : "");
      } catch (err) {
        console.error(err);
      }
      ask();
    });
  };

  ask();
}

main();

File: java/src/main/java/aibuffet/Chatbot.java
package aibuffet;

import com.anthropic.client.AnthropicClient;
import com.anthropic.client.okhttp.AnthropicOkHttpClient;
import com.anthropic.models.messages.Message;
import com.anthropic.models.messages.MessageCreateParams;
import com.anthropic.models.messages.Model;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Scanner;

/**
 * Simple chat loop using the Anthropic Java SDK.
 * Model name is not passed directly to the client constructor, to showcase
 * detection of model usage via code flow.
 */
public final class Chatbot {

    private static final String ENV_KEY = "ANTHROPIC_API_KEY";

    private static AnthropicClient initClient() {
        String apiKey = System.getenv(ENV_KEY);
        if (apiKey == null || apiKey.isBlank()) {
            System.err.println(ENV_KEY + " not set in environment.");
            System.exit(1);
        }
        return AnthropicOkHttpClient.fromEnv();
    }

    private static Model resolveModel() {
        return Model.CLAUDE_3_5_SONNET_20240620;
    }

    public static void main(String[] args) {
        AnthropicClient client = initClient();
        Model model = resolveModel();
        System.out.println("Chat with Claude (type 'exit' to quit).\n");

        Scanner scanner = new Scanner(new BufferedReader(new InputStreamReader(System.in)));
        while (true) {
            System.out.print("> ");
            if (!scanner.hasNextLine()) {
                break;
            }
            String prompt = scanner.nextLine();
            if ("exit".equals(prompt)) {
                break;
            }

            try {
                MessageCreateParams params = MessageCreateParams.builder()
                        .model(model)
                        .maxTokens(1024L)
                        .addUserMessage(prompt)
                        .build();
                Message response = client.messages().create(params);
                response.content().stream()
                        .flatMap(block -> block.text().stream())
                        .forEach(text -> System.out.println(text.text()));
            } catch (Exception err) {
                System.err.println(err.getMessage());
            }
        }
    }
}
